nvdaHelper/remote/log.cpp

```diff
@@ -12,13 +12,56 @@ This license can be found at:
 http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
 */
 
+#include <string>
+#include <queue>
 #include <crtdbg.h>
 #include "nvdaControllerInternal.h"
+#include "nvdaHelperRemote.h"
 #include <common/log.h>
 
+std::deque<std::tuple<int, std::wstring>> logQueue;
+
+// Fetch a  maximum number of messages from the log queue
+// and send them onto NvDA via rpc.
+void __stdcall log_flushQueue(ULONG_PTR maxCount) {
+	while(!logQueue.empty()) {
+		if(maxCount < 1) break;
+		{
+			auto& [level, msg] = logQueue.front();
+			nvdaControllerInternal_logMessage(level, GetCurrentProcessId(), msg.c_str());
+		}
+		logQueue.pop_front();
+		--maxCount;
+	}
+}
+
 void logMessage(int level, const wchar_t* msg) {
-	if(level>LOGLEVEL_DEBUG) nvdaControllerInternal_logMessage(level,GetCurrentProcessId(),msg);
+	// Always log to any connected debugger 
 	OutputDebugString(msg);
+	if(
+		!inprocMgrThreadHandle
+		|| GetCurrentThreadId() != GetThreadId(inprocMgrThreadHandle)
+	) {
+		// either the NVDA inproc manager thread is not yet running,
+		// Or this message is being logged from outside NVDA's inproc manager thread.

```

Comment:
I may be missing something here. If "the NVDA inproc manager thread is not yet running" I.E. `inprocMgrThreadHandle == nullptr` then this message will never be flushed, or perhaps it will, but only once something else is logged from the inproc manager thread.